flake.nix

```diff
@@ -0,0 +1,180 @@
+        rust-src = pkgs.lib.cleanSourceWith {
+          src = pkgs.lib.cleanSource ./.;
+          filter = pkgs.nix-gitignore.gitignoreFilterPure
+            (name: type:
+              # nix files are not used as part of build
+              (
+                (type == "regular" && pkgs.lib.strings.hasSuffix ".nix" name)
+                == false
+                &&
+                (type == "directory" && ".github" == name) == false
+              )
+            )
+
+
+            [ ./.gitignore ] ./.;
+        };
+
+        golden-gate-runtime = craneLib.buildPackage (common-wasm-attrs // {
+          installPhase = ''
+            mkdir --parents $out/lib
+            cp ./target/wasm32-unknown-unknown/release/wbuild/golden-gate-runtime/golden_gate_runtime.compact.compressed.wasm $out/lib
+          '';
+          src = rust-src;
+          cargoArtifacts = common-wasm-deps;
+        });
+      in
+      rec {
+        packages = flake-utils.lib.flattenTree {
+          inherit golden-gate-runtime;
+          default = golden-gate-runtime;
+        };
+
+
+        devShells = {
+          default = devenv.lib.mkShell {
+            inherit inputs pkgs;
+            modules =
+              let
+
+                dylib = {
+                  buildInputs = with pkgs; [ openssl ] ++ darwin;
+                  nativeBuildInputs = with pkgs;[ pkg-config ];
+                  doCheck = false;
+                };
+                rust-deps = pkgs.makeRustPlatform {
+                  inherit pkgs;
+                  # dylint needs nightly
+                  cargo = pkgs.rust-bin.beta.latest.default;
+                  rustc = pkgs.rust-bin.beta.latest.default;
+                };
+                cargo-dylint = with pkgs; rust-deps.buildRustPackage (rec {
+                  pname = "cargo-dylint";
+                  version = "2.1.5";
+                  src = fetchCrate {
+                    inherit pname version;
+                    sha256 = "sha256-kH6dhUFaQpQ0kvzNyLIXjFAO8VNa2jah6ZaDO7LQKO0=";
+                  };
+
+                  cargoHash = "sha256-YvQI3H/4eWe6r2Tg8qHJqfnw/NpuGHtkRuTL4EzF0xo=";
+                  cargoDepsName = pname;
+                } // dylib);
+                dylint-link = with pkgs; rust-deps.buildRustPackage (rec {
+                  pname = "dylint-link";
+                  version = "2.1.5";
+                  src = fetchCrate {
+                    inherit pname version;
+                    sha256 = "sha256-oarEYhv0i2wAPmahx0vgWN3kmfEsK3s6D3+qkOqF9pc=";
+                  };
+
+                  cargoHash = "sha256-pMr9hddHAIyIclHRpxqdUaHphjSAVDnvfNjWGDA2EM4=";
+                  cargoDepsName = pname;
+                } // dylib);
+                # can `cargo-contract` and nodejs ui easy here 
+              in
+              [
+                {
+                  packages = with pkgs;[ rust-toolchain binaryen clang llvmPackages.bintools dylint-link ] ++ darwin;
```

Comment:
For me, it doesn't compile if I don't add gnumake here.

Reply:
Also, I have checked, and dev-shell can't compile contracts. I have tried to add cargo contracts but failed due to a lack of experience with nix. I would appreciate it if you would adjust the dev env.